### PR TITLE
IVT computation app

### LINKED ISSUE #391: IVT calculation app for HighResMIP
We need an app for calculating integrated vapor transport from the HighResMIP models.

It will need to accept files containing both 4D (time, lev, lat ,lon) and 3D (time, lat, lon) variables:

- 4D variables
  - humidity Q
  - zonal wind U
  - meridional wind V
- 3D variables
  - surface humidity QSFC
  - surface zonal wind USFC
  - surface meridional wind VSFC
  - surface pressure PS, or
    - surface sea-level pressure
    - surface elevation (a 2D [lat, lon] field)

It will output netCDF files containing IVT, a 3D field that contains the vertical integral of horizontal moisture transport.

A typical run command should be something like:
```bash
teca_ivt_integral \
    --q_file_regex QFILEREGEX \
    --u_file_regex UFILEREGEX \
    --v_file_regex VFILEREGEX \
    --qsfc_file_regex QSFCFILEREGEX \
    --usfc_file_regex USFCFILEREGEX \
    --vsfc_file_regex VSFCFILEREGEX \
    --ps_file_regex PSFILEREGEX
    --q_var_name QVARNAME \
    --u_var_name UVARNAME \
    --v_var_name VVARNAME \
    --ps_var_name PSVARNAME \
    --output_ivt_var_name IVTVARNAME \
    --output_file_name OUTPUTFILETEMPLATE \
```

The pipeline will be something like:
```
CF Reader --> Normalize Coordinates --> Moisture Transport vector (Q*U, V*Q) --> Vertical Integral (uIVT, vIVT) --> l2_norm (IVT) --> CF writer
```

Depends on:

- #353 
- #390

### PR DESCRIPTION
**Pull request by elbashandy**
Resolves #391

### CODE REVIEW

File: apps/CMakeLists.txt
Comment (burlen) [inline apps/CMakeLists.txt:22]: let's change the name to `teca_integrated_vapor_transport`

File: apps/teca_ivt_computation.cpp
Comment (burlen) [inline apps/teca_ivt_computation.cpp:61]: not needed

**Review by burlen — CHANGES_REQUESTED** [review]
Looking good but I see a few things that should be changed.

**Comment by elbashandy** [comment]
@burlen Thanks for reviewing this. I think I covered all the required changes. Please give it a look when you have a chance.

File: apps/teca_cartesian_mesh_diff.cpp
Comment (elbashandy) [reply apps/teca_cartesian_mesh_diff.cpp:34]: Yes I must have made a mistake while rebasing. Just checked, everything looks in place.

**Review by burlen — CHANGES_REQUESTED** [review]
a few changes to be made

File: test/apps/test_integrated_vapor_transport_app.sh
Comment (burlen) [reply test/apps/test_integrated_vapor_transport_app.sh:32]: confirmed. teca_cartesian_mesh diff doesn't use the index executive as a result an empty request is always made, hence step 0 is always requested and no arrays are being compared. this also explains why it didn't matter which order the reference and test datasets were passed on the command line. Added an issue for this #498

**Review by burlen — APPROVED** [review]
all set, but depends on #499. hold off until that is merged

**Pull request merged**
